### PR TITLE
Fix Pearson QTI Issues

### PR DESCRIPTION
**Pull request by ashishkumarqait**
Fix Details: following tasks have been performed.

Task 1 : Collect html from html file's (defined by "object" tag) body tag and add them at respective places in exported learnosity JSON file.

Task 2 : Collect all **feedbackInline**'s content in metadata as follows (done)

"metadata": { "distractor_rationale_response_level": [ "", "distractor feedback 1", "distractor feedback 2", "distractor feedback 3" ] },

### CODE REVIEW

File: src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php [hunk 1]
```diff
@@ -90,4 +99,31 @@ private function buildOptions(SimpleChoiceCollection $simpleChoices)
...
+            if(property_exists($flow,'dataPlaceHolder')){
+                $property = $class->getProperty('dataPlaceHolder');
+                $property->setAccessible(true);
+                $feed = $property->getValue($flow);
+                $feedArray = array_values((array) $feed);
+                if(sizeof($feedArray)==2){
+                    $feeddataArray = array_values((array) $feedArray[0]);
+                    $feedarr = array_values((array) $feeddataArray[3][0]);
+                    if($feedarr[2]=='text/html'){
+                        $htmlfile = ConvertToLearnosityService::$inputDir.'/'.$feedarr[1];
```
Comment (ttton) [inline src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php]: See my comment on the approach here: [#diff-6f0e68dc90c0979d95f5acaf95f64b4eR52](https://github.com/Learnosity/learnosity-qti/pull/29/files#diff-6f0e68dc90c0979d95f5acaf95f64b4eR52)

File: src/Utils/QtiMarshallerUtil.php [hunk 2]
```diff
@@ -74,7 +89,7 @@ public static function marshall(QtiComponent $component)
         $node = $dom->importNode($element, true);
 
         $string = $dom->saveXML($node);
-
+
```
Comment (ttton) [inline src/Utils/QtiMarshallerUtil.php]: PSR-2:
* Remove added whitespace on newline

File: src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php [hunk 3]
```diff
@@ -90,4 +99,31 @@ private function buildOptions(SimpleChoiceCollection $simpleChoices)
...
     }
+
+    private function buildFeedbackMetadata(SimpleChoiceCollection $simpleChoices)
+    {
+        /* @var $choice SimpleChoice */
+        $metadata = [];
+        foreach ($simpleChoices as $key => $choice) {
+            $flow = $choice->getContent();
+            $class = new \ReflectionClass(get_class($flow));
+            if(property_exists($flow,'dataPlaceHolder')){
```
[the remark this answers is not in the document]
Comment (pooja31chaudhary) [reply src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php]: Done . Please check PR #41

File: src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php [hunk 4]
```diff
@@ -90,4 +99,31 @@ private function buildOptions(SimpleChoiceCollection $simpleChoices)
...
+                $property = $class->getProperty('dataPlaceHolder');
+                $property->setAccessible(true);
+                $feed = $property->getValue($flow);
+                $feedArray = array_values((array) $feed);
+                if(sizeof($feedArray)==2){
```
[the remark this answers is not in the document]
Comment (pooja31chaudhary) [reply src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php]: Done . Please check PR #41

File: src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php [hunk 5]
```diff
@@ -90,4 +99,31 @@ private function buildOptions(SimpleChoiceCollection $simpleChoices)
...
+                    $feeddataArray = array_values((array) $feedArray[0]);
+                    $feedarr = array_values((array) $feeddataArray[3][0]);
+                    if($feedarr[2]=='text/html'){
```
[the remark this answers is not in the document]
Comment (pooja31chaudhary) [reply src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php]: Done . Please check PR #41

File: src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php [hunk 6]
```diff
@@ -90,4 +99,31 @@ private function buildOptions(SimpleChoiceCollection $simpleChoices)
...
+                        $htmlfile = ConvertToLearnosityService::$inputDir.'/'.$feedarr[1];
+                        $metadata[] = HtmlExtractorUtil::getHtmlData(realpath($htmlfile));
+                    }
+                }else{
```
[the remark this answers is not in the document]
Comment (pooja31chaudhary) [reply src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php]: Done . Please check PR #41

File: src/Services/ConvertToLearnosityService.php [hunk 7]
```diff
@@ -48,9 +48,12 @@ class ConvertToLearnosityService
     protected $useResourceIdentifier      = false;
 
     private $assetsFixer;
+    
+    public static $inputDir;
 
     public function __construct($inputPath, $outputPath, OutputInterface $output, $organisationId)
     {
+        self::$inputDir       = $inputPath;
```
[the remark this answers is not in the document]
Comment (pooja31chaudhary) [reply src/Services/ConvertToLearnosityService.php]: Done . Please check PR #41

File: src/Utils/QtiMarshallerUtil.php [hunk 8]
```diff
@@ -55,10 +58,22 @@ public static function marshallValidCollection(QtiComponentCollection $collectio
     public static function marshallCollection(QtiComponentCollection $collection)
     {
         $results = [];
-        foreach ($collection as $component) {
-            $results[] = self::marshall($component);
+        foreach ($collection as $component){
```
[the remark this answers is not in the document]
Comment (pooja31chaudhary) [reply src/Utils/QtiMarshallerUtil.php]: Done . Please check PR #41

File: src/Utils/HtmlExtractorUtil.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+class HtmlExtractorUtil {
+
+    /**
+     * Get html body contents from a given html file
+     * 
+     * @param string $file filepath to read html
+     * @return string
+     */
+    public static function getHtmlData($file) {
+        $file = getenv('GET_INPUT_PATH') . '/' . $file;
```
[the remark this answers is not in the document]
Comment (pooja31chaudhary) [reply src/Utils/HtmlExtractorUtil.php]: It has removed. please check PR #41

File: src/Services/ConvertToLearnosityService.php [hunk 10]
```diff
@@ -48,9 +48,12 @@ class ConvertToLearnosityService
     protected $useResourceIdentifier      = false;
 
     private $assetsFixer;
+    
+    public static $inputDir;
```
[the remark this answers is not in the document]
Comment (pooja31chaudhary) [reply src/Services/ConvertToLearnosityService.php]: Done . Please check PR #41

File: src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php (hunk 1 above)
Comment (pooja31chaudhary) [reply src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php]: Done . Please check PR #41

File: src/Utils/QtiMarshallerUtil.php (hunk 2 above)
Comment (pooja31chaudhary) [reply src/Utils/QtiMarshallerUtil.php]: Done . Please check PR #41

**Comment by pooja31chaudhary** [comment]
All above feed backs for this PR has been fixed and a new PR #41 have been generated. So please check PR #41.

**Comment by jhume89** [comment]
Hi @pooja31chaudhary,
Thanks for fixing those things, but please push new changes on top of the original PR (do not create a new one - that just loses all the history and continuity of the review we've already done).

This should be the standard workflow moving forward. Please push your changes from #41 onto this branch and we'll continue our review here.

**Comment by ttton** [comment]
Closing this PR in favour of #41.

**Pull request closed without merging**